libs/stats/tests/test_pq_bitmask.py

```diff
@@ -0,0 +1,162 @@
+def dataset_with_aerosol_band(dataset):
+    aerosol_mask = 0b1100_0000
+    no_data = 0b0000_0001
+    band_aerosol = np.array([
+        [[0, 0], [aerosol_mask, no_data]],
+        [[no_data, 0], [0, 0]],
+        [[aerosol_mask, 0], [0, 0]],
+    ])
+
+    band_aerosol = da.from_array(band_aerosol, chunks=(3, -1, -1))
+
+    dataset["SR_QA_AEROSOL"] = (("spec", "y", "x"), band_aerosol)
+    return dataset
+
+def test_native_transform(dataset):
+    pq = StatsPQLSBitmask()
+
+    xx = pq._native_tr(dataset)
+    tr_result = xx.compute()
+
+    expected_result = np.array([
+        [[True, True], [True, False]],
+        [[False, True], [True, True]],
+        [[True, True], [True, True]],
+    ])
+    keeps = tr_result["keeps"].data
+    assert (keeps == expected_result).all()
+
+    expected_result = np.array([
+        [[True, True], [False, True]],
+        [[True, True], [True, True]],
+        [[True, False], [False, True]],
+    ])
+    clear = tr_result["clear"].data
+    assert (clear == expected_result).all()
+
+def test_fuser(dataset):
+    pq = StatsPQLSBitmask()
+
+    xx = pq._native_tr(dataset)
+    xx = xx.groupby("solar_day").map(pq._fuser)
+    fuser_result = xx.compute()
+
+    expected_result = np.array(
+        [[True, True], [True, True]],
+    )
+    clear = fuser_result["clear"].data
+    assert (clear == expected_result).all()
+
+def test_reduce(dataset):
+    pq = StatsPQLSBitmask(filters=[[1,1,0]])
+
+    xx = pq._native_tr(dataset)
+    xx = pq.reduce(xx)
+    reduce_result = xx.compute()
+
+    assert set(reduce_result.data_vars.keys()) == set(
+        ["total", "clear", "clear_1_1_0"]
+    )
+
+    expected_result = np.array(
+        [[2, 3], [3, 2]]
+    )
+    total = reduce_result["total"].data
+    assert (total == expected_result).all()
+
+    expected_result = np.array(
+        [[2, 2], [1, 2]]
+    )
+    clear = reduce_result["clear"].data
+    assert (clear == expected_result).all()
+
+    expected_result = np.array(
+        [[1, 2], [2, 1]]
+    )
+    clear_1_1_0 = reduce_result["clear_1_1_0"].data
+    assert (clear_1_1_0 == expected_result).all()
+
+def test_native_transform_for_clear_aerosol(dataset_with_aerosol_band):
+    pq = StatsPQLSBitmask(pq_band = "QA_PIXEL", aerosol_band = "SR_QA_AEROSOL")
+
+    xx = pq._native_tr(dataset_with_aerosol_band)
+    tr_result = xx.compute()
+
+    expected_result = np.array([
+        [[True, True], [False, True]],
+        [[True, True], [True, True]],
+        [[False, True], [True, True]],
+    ])
+    clear_aerosol = tr_result["clear_aerosol"].data
+    assert (clear_aerosol == expected_result).all()
+
+def test_fuse_for_clear_aerosol(dataset_with_aerosol_band):
+    pq = StatsPQLSBitmask(pq_band = "QA_PIXEL", aerosol_band = "SR_QA_AEROSOL")
+
+    xx = pq._native_tr(dataset_with_aerosol_band)
+    xx = xx.groupby("solar_day").map(pq._fuser)
+    fuser_result = xx.compute()
+
+    expected_result = np.array(
+        [[True, True], [True, True]],
```

Comment:
Maybe this should be:

```suggestion
        [[False, True], [False, True]],
```